handbook/ldap/ldap.md

```diff
@@ -0,0 +1,118 @@
+
+    **Structure:** `ldap://<server-dns>:<port>/<target-entry-dn>?<username-attribute>`
+
+    **Example:** `ldap://example.com:389/dc=example,dc=com?sAMAccountName`
+
+    In this example the server is `example.com`, the port is 389, the base search DN is `dc=example,dc=com` and the LDAP attribute used for user login is `sAMAccountName`. 
+
+    <figure>
+      <img src="../../img/ldap-login.png" width="500" />
+      <figcaption>Username `jdoe` is used because the `sAMAccountName` attribute is specified in the LDAP URL.</figcaption>
+    </figure>
+
+If a username attribute is not specified **NI Web Server Configuration** will automatically add `uid` as the attribute. Depending on how your LDAP directory is setup this attribute may or may not be available.
+
+!!! note "Common username attributes"
+    - `uid`
+    - `userPrincipalName`
+    - `sAMAccountName`
+
+## Bind User
+
+The bind user can be specified by providing either a `distinguishedName` or a `userPrincipalName`.
+
+!!! note "Bind user formats for the user Jane Doe"
+    **distinguishedName:** cn=jdoe,dc=example,dc=com
+
+    **userPrincipalName:** jane.doe@example.com
+
+## Mapping LDAP Attributes and Groups to SystemLink Workspaces and Roles
+
+To add a user or collection of users to a workspace and assign a role you must complete the role mapping workflow.
+
+1. Login into the SystemLink web application with a user mapped to the [**Server Administrator**](https://www.ni.com/documentation/en/systemlink/latest/setup/predefined-roles/) role.
+
+2. Go to **Security** > **Workspaces** and click the gear icon in the top right.
+
+3. Create a new workspace or edit an existing workspace.
+
+4. Go to the **Role Mappings** tab
+
+5. Click **+MAPPING** and select one of the available LDAP mapping types: *LDAP Group*, *LDAP User*, or *LDAP Attribute*.
+
+6. If you have have selected *LDAP Attribute* enter a valid key/value. For *LDAP User* or *LDAP Group* enter a valid value.
+
+### LDAP Group Mapping
+
+The LDAP Group mapping looks for the values in the LDAP `memberOf` attribute. This attribute's value is an array of group DNs. The value provided to SystemLink must exactly match one value available in the `memberOf` attribute array.
+
+### LDAP User Mapping
+
+Similar to the LDAP attribute that determines what users use for their username during login, the value for LDAP user mapping must match the user ID attribute specified in the LDAP URL.
+
+!!! note "Example LDAP user mappings with different LDAP URLs for the user Jane Doe"
+
+    **URL:** `ldap://example.com/dc=example,dc=com?sAMAccountName`
+    **LDAP User:** jdoe
+
+     **URL:** `ldap://example.com/dc=example,dc=com?userPrincipalName`
+    **LDAP User:** jane.doe@example.com
+
+### LDAP Attribute Mapping
+
+Any available LDAP attribute may be used to create a workspace and role mapping. In this case both the name and value of the attribute must match exactly for the mapping to be successful.
```

Comment:
>Any available LDAP attribute may be used to create a workspace and role mapping.

You can use any available LDAP attribute to create a workspace and role mapping.

>In this case both the name and value of the attribute must match exactly for the mapping to be successful.

So this isn't always the case? It doesn't always have to be an exact match?

Reply:
This is the case in this instance because the user has to specify the key (LDAP attribute) and its value. In cases of LDAP user of LDAP group they only need to match the value because we do the key for them.